Visualization/CreateBlankResults.py

```diff
@@ -5,88 +5,109 @@
-filesToReplace = sys.argv[1]
-geneList = sys.argv[2]
-
-# File extension check
-if not filesToReplace.endswith(".txt"):
-    print("[Warning] " + geneList + " may not be a text file!")
-
-# File extension check
-if not geneList.endswith(".faa"):
-    print("[Warning] " + geneList + " may not be a FASTA file!")
-
-# Reads sequence file list and stores it as a string.
-try:
-    with open(filesToReplace, "rU") as newFile:
-        filesToReplace = newFile.read()
-        newFile.close()
-except IOError:
-    print("Failed to open " + filesToReplace)
-    exit(1)
-
-# Splits string into a list. Each element is a single line from the string.
-filesToReplaceList = filesToReplace.splitlines()
-
-FakeResults = []
-try:
-    handle = open(geneList, "rU")
-    SeqRecords = SeqIO.parse(handle, "fasta")
-    for record in SeqRecords:
-        FakeResults.append(
-            record.id + ",sseqid,0,evalue,qcovhsp,bitscore")  # qseqid sseqid pident evalue qcovhsp bitscore
-    handle.close()
-except IOError:
-    print("Failed to open " + filesToReplace + " or " + geneList)
-    exit(1)
-
-FakeResultsOut = "\n".join(FakeResults)
-
-print(">> Writing Fake Results...")
-
-for x in filesToReplaceList:
-    replacementFileName = x + ".csv"
-
-    try:
-        writeFile = open(replacementFileName, "w")
-        writeFile.write(FakeResultsOut)
-        writeFile.close()
-    except IOError:
-        print("Failed to create " + replacementFileName)
-        exit(1)
-
-print(">> Done.")
+# Checks whether or not the provided input_csv is an empty file. Returns logical.
+def check_if_input_csv_is_empty(input_csv):
+    if os.stat(input_csv).st_size == 0: # Returns True is the CSV is empty and False if not
+        return True
+    else:
+        return False
+
+
+# Checks whether or not the provided file has the expected extension. Throws warning if not. No return.
+def file_extension_check(filename, extension):
+    if not filename.endswith('.' + extension):
+        logger.warning("'" + filename + "' may not be a '" + extension + "' file!")
+
+
+# Uses input query_proteins FAA file to build a blank BLAST CSV table. Returns the table.
+def generate_blank_results(query_proteins):
+    with open(query_proteins, "rU") as fasta_file:
+        fasta_entries = SeqIO.parse(fasta_file, "fasta")
+
+    blank_results_list = []
+    for entry in fasta_entries:
+        blank_results_list.append(
+            entry.id + ",NA,NA,NA,NA,NA")  # qseqid sseqid pident evalue qcovhsp bitscore
+
+    blank_results = "\n".join(blank_results_list)
+    return(blank_results)
+
+
+# Writes a blank_results table to a file with name new_blast_file. No return.
+def write_blank_results(blank_results, new_blast_file):
+    with open(new_blast_file, "w") as writeFile:
+        writeFile.write(blank_results)
+
+
+def main(args):
+    # Set up the logger
+    logging.basicConfig(level=logging.DEBUG, format="[ %(asctime)s UTC ]: %(levelname)s: %(module)s: %(message)s")
```

Comment:
@jmtsuji In the future we may want to make the logging level toggleable via a CLI flag.